store/store.js

```diff
@@ -45,9 +44,16 @@ class RootStore {
   onDataProviderChange = () =>
     autorun(() => {
       if (isServer) return
-      const route = new Route(window.location.pathname)
-      route.dataProvider = this.dataProvider
-      Router.push(route.href, route.as)
+
+      // if user has no access to dataProvider redirect them
+      const route = new Route(Router.asPath)
+      if (
+        route.dataProvider &&
+        parseInt(route.dataProvider, 10) !== this.user.organisationId
```

Comment:
One more line with comparison an organisation to a data provider

Reply:
I don't get it.

Reply:
You compare a data provider ID to an organisation ID. These entities are 2 different so comparison is wrong.